From 04ee59f8914f48a9f6fca348ef0582798f37efa8 Mon Sep 17 00:00:00 2001
From: Liu Zhiquan <zhiquan.liu@intel.com>
Date: Thu, 20 Jun 2019 13:23:13 +0800
Subject: [PATCH 3/3] layer damage region din't set to composer service

In setSurfaceDamage function, this redundant call will be optimized
if the damage region('damage') is same as prior damage region
('mDamageRegion'). Rect::INVALID_RECT indicates full screen damaged.
If layer first 'damage' region is Rect::INVALID_RECT, it hase the same
value with 'mDamageRegion' default value, and the first damage call
will treat as redundant call.
We change 'mDamageRegion' default value to EMPTY_RECT, which indicates
the screen don't need be update, then neccesary damage call will send
to composer service

Change-Id: I72f03651d2f1889459b187fc24fe8f59f01fd0b6
Tracked-On:
Signed-off-by: Liu Zhiquan <zhiquan.liu@intel.com>
---
 services/surfaceflinger/DisplayHardware/HWC2.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/surfaceflinger/DisplayHardware/HWC2.h b/services/surfaceflinger/DisplayHardware/HWC2.h
index b7cdf7f6c..d254caa63 100644
--- a/services/surfaceflinger/DisplayHardware/HWC2.h
+++ b/services/surfaceflinger/DisplayHardware/HWC2.h
@@ -442,7 +442,7 @@ private:
     // Cached HWC2 data, to ensure the same commands aren't sent to the HWC
     // multiple times.
     android::Region mVisibleRegion = android::Region::INVALID_REGION;
-    android::Region mDamageRegion = android::Region::INVALID_REGION;
+    android::Region mDamageRegion = android::Region(android::Rect::EMPTY_RECT);
     android::ui::Dataspace mDataSpace = android::ui::Dataspace::UNKNOWN;
     android::HdrMetadata mHdrMetadata;
     android::mat4 mColorMatrix;
--
2.21.0
